test/integ/test_connection.py

```diff
@@ -1107,3 +1108,54 @@ def test_ocsp_cache_working(conn_cnx):
     with conn_cnx() as cnx:
         assert cnx
     assert OCSP_CACHE.telemetry["hit"] + OCSP_CACHE.telemetry["miss"] > original_count
+
+
+@pytest.mark.skipolddriver
+def test_imported_packages_telemetry(db_parameters, capture_sf_telemetry):
+    try:
+        config = {
+            "user": db_parameters["user"],
+            "password": db_parameters["password"],
+            "host": db_parameters["host"],
+            "port": db_parameters["port"],
+            "account": db_parameters["account"],
+            "schema": db_parameters["schema"],
+            "database": db_parameters["database"],
+            "protocol": db_parameters["protocol"],
+            "timezone": "UTC",
+        }
+
+        # these imports are not used but for testing
+        import html.parser  # noqa: F401
+        import json  # noqa: F401
+        import multiprocessing as mp  # noqa: F401
+        from datetime import date  # noqa: F401
+        from math import sqrt  # noqa: F401
+
+        with snowflake.connector.connect(**config) as conn:
+            with capture_sf_telemetry.patch_connection(conn, False) as telemetry_test:
+                conn._log_telemetry_imported_packages(conn._imported_vars)
```

Comment:
that's the best way I can think of to test telemetry based on the current test infra (I'm not able to get the telemetry sent during connection creation). Please comment if you have a better way.

Reply:
Yea I think based on the complexity of this, it may be easier to just get the packages in the python environment and send that as telemetry. What do you think @sfc-gh-achandrasekaran